common/internal_model/src/test/scala/uk/ac/wellcome/models/work/internal/InstantRangeTest.scala

```diff
@@ -0,0 +1,45 @@
+package uk.ac.wellcome.models.work.internal
+
+import org.scalatest.prop.TableDrivenPropertyChecks._
+import org.scalatest.{FunSpec, Matchers}
+import uk.ac.wellcome.models.work.generators.InstantRangeGenerators
+
+class InstantRangeTest
+    extends FunSpec
+    with Matchers
+    with InstantRangeGenerators {
+
+  it("parses valid date labels") {
+    val instantRange = createInstantRangeWith(
+      "1909",
+      "1909-01-01T00:00",
+      "1909-12-31T23:59:59.999999999",
+      inferred = false)
+
+    val inferredInstantRange = createInstantRangeWith(
+      "[2123]",
+      "2123-01-01T00:00",
+      "2123-12-31T23:59:59.999999999",
+      inferred = true)
+
+    forAll(
+      Table(
+        ("label", "parsedInstantRange"),
+        ("1909", instantRange),
+        ("[2123]", inferredInstantRange),
+      )) { (label, parsedInstantRange) =>
+      InstantRange.parse(label) shouldBe Some(parsedInstantRange)
+    }
+  }
+
+  it("does not parse invalid date labels") {
+    forAll(
+      Table(
+        "label",
+        "nineteen sixty what, nineteen sixty who",
+        "[21233]",
```

Comment:
other possible test cases:

* four characters, not all digits
* four characters in square brackets, not all digits
* less than four characters, with and without square brackets